### PR TITLE
Activate flake8 for Python 3

### PR DESCRIPTION
**Pull request by nth10sd**
These are the remaining changes needed for funfuzz to be forward-compatible with Python 3.

### CODE REVIEW

File: js/compileShell.py [hunk 1]
```diff
@@ -167,7 +167,9 @@ def ensureCacheDir():
     # Expand long Windows paths (overcome legacy MS-DOS 8.3 stuff)
     # This has to occur after the shell-cache directory is created
     if sps.isWin:  # adapted from http://stackoverflow.com/a/3931799
-        winTmpDir = unicode(cacheDir)
+        # Python 3 no longer has the unicode function, see http://python-future.org/compatible_idioms.html#unicode
+        from builtins import str as utext
```
Comment (jschwartzentruber) [inline js/compileShell.py]: `builtins` isn't stdlib in python2. Are you adding a dependency?
What I did in avalanche is this:

    from __future__ import unicode_literals
    if sys.version_info.major == 2:
        # pylint: disable=redefined-builtin,invalid-name
        str = unicode

This changes all old strings to unicode, and if you need raw strings use `b""` or `bytes()` instead (which work in both py2.7 & 3).

File: js/compileShell.py [hunk 2]
```diff
@@ -578,6 +578,7 @@ def fixateVer(pcfile):
         """Returns the current version number (47.0a2)."""
         with open(pcfile, 'rb') as f:
             for line in f:
+                line = line.decode("utf-8", "replace")
```
Comment (jschwartzentruber) [inline js/compileShell.py]: You could also accomplish this using the `io` module:

    with io.open(pcfile, encoding="utf-8", errors="replace"):

Comment (nth10sd) [inline js/compileShell.py]: Nice, I didn't know about this! Thanks!

Comment (nth10sd) [inline js/compileShell.py]: I kept the 'rb' mode for [Windows compatibility](https://docs.python.org/2/tutorial/inputoutput.html#reading-and-writing-files).

Comment (jschwartzentruber) [inline js/compileShell.py]: Did you try this? I would have guessed `io.open()` would work on windows ok (with `encoding` and `errors` specified).

Comment (nth10sd) [inline js/compileShell.py]: What I meant was, I [switched to](https://github.com/MozillaSecurity/funfuzz/pull/119/commits/da871d5e1dabedb989e93e44261946cffd1d7f53) use io.open but with 'rb' mode:

```
with io.open(pcfile, mode='rb', encoding="utf-8", errors="replace")
```

Comment (nth10sd) [inline js/compileShell.py]: Oh wait, you're right. With encoding specified, it should not be in bytes mode, both in [Python 2](https://docs.python.org/2/library/io.html#io.open) and 3:

```
Traceback (most recent call last):
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 743, in <module>
    main()
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 738, in main
    obtainShell(shell, updateToRev=options.revision)
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 663, in obtainShell
    cfgJsCompile(shell)
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 236, in cfgJsCompile
    compileJs(shell)
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 499, in compileJs
    version = extractVersions(shell.getJsObjdir())
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 592, in extractVersions
    return fixateVer(jspcNewFilename)
  File "/home/gkwubu/funfuzz/js/compileShell.py", line 583, in fixateVer
    with io.open(pcfile, mode='rb', encoding="utf-8", errors="replace") as f:
ValueError: binary mode doesn't take an encoding argument
```

Fixed in rev ec986974eed7110d3ca64624dd62e73c7746f5e4.

**Pull request merged**